tests/functional/workloads/cnv/test_vm_hotplug_unplug.py

```diff
@@ -0,0 +1,120 @@
+class TestVmHotPlugUnplug(E2ETest):
+    """
+    Test case for VM hot plugging and unplugging of PVC disks.
+    """
+
+    def test_vm_hot_plugging_unplugging(
+        self, project_factory, multi_cnv_workload, setup_cnv
+    ):
+        """
+        Verify that hotplugging and hot unplugging of a PVC to/from a VM works
+
+            Steps:
+            1. Hotplug disk to the running VM based on PVC.
+            2. Verify the disk is attached to VM
+            3. Add data to disk
+                a. Identify newly attached disk.
+                b. Create file or do dd on new disk
+            4. Reboot the VM,
+            5. After reboot check if disk is still attached.
+            6. Make sure newly added data on new vm disk is intact
+            7. Unplug(Dettach) the disk from vm
+            8. Verify disk is successfully detached using console or cli
+            9. login into VM and confirm disk is no longer listed.
+            10 Repeat the above tests for DVT based VM
+        """
+
+        # Create project and get VM details
+        proj_obj = project_factory()
+        file_paths = ["/file.txt", "/new_file.txt"]
+        vm_objs_def, vm_objs_aggr, _, _ = multi_cnv_workload(
+            namespace=proj_obj.namespace
+        )
+        vm_list = vm_objs_def + vm_objs_aggr
+
+        log.info(f"Total VMs to process: {len(vm_list)}")
+        for index, vm_obj in enumerate(vm_list):
+            # Create a new PVC for hotplug
+            pvc_obj = create_pvc(
+                sc_name=constants.DEFAULT_CNV_CEPH_RBD_SC,
+                namespace=vm_obj.namespace,
+                size="20Gi",
+                access_mode=constants.ACCESS_MODE_RWX,
+                volume_mode=constants.VOLUME_MODE_BLOCK,
+            )
+            log.info(f"PVC {pvc_obj.name} created successfully")
+
+            # List disks before attaching the new volume
+            before_disks = vm_obj.run_ssh_cmd(
+                command="lsblk -o NAME,SIZE,MOUNTPOINT -P"
+            )
+            log.info(f"Disks before hotplug:\n{before_disks}")
+
+            # Hotplug the PVC to the VM
+            vm_obj.addvolume(volume_name=pvc_obj.name)
+            assert verifyvolume(
+                vm_obj, volume_name=pvc_obj.name
+            ), f"Unable to found volume {pvc_obj.name} mounted on VM: {vm_obj.name}"
+            log.info(f"Hotplugged PVC {pvc_obj.name} to VM {vm_obj.name}")
+
+            # List disks after attaching the new volume
+            after_disks = vm_obj.run_ssh_cmd("lsblk -o NAME,SIZE,MOUNTPOINT -P")
+            log.info(f"Disks after hotplug:\n{after_disks}")
+
+            # Identify the newly attached disk
+            new_disks = set(after_disks) - set(before_disks)
+            log.info(f"Newly attached disks: {set(new_disks)}")
+
+            # Perform I/O operation on the new disk
+            log.info(
+                f"Running I/O operation on the newly attached disk in VM {vm_obj.name}"
+            )
```

Comment:
from the log message, it looks like you are running IO to the newly attached disk. How are you making sure that the created file indeed goes to the new disk?